api/app/morecast_v2/forecasts.py

```diff
@@ -75,36 +80,33 @@ async def construct_wf1_forecasts(session: ClientSession, forecast_records: List
     for forecast in forecast_records:
         forecast_timestamp = datetime.fromtimestamp(forecast.for_date / 1000, timezone.utc)
         # Check if an existing daily was retrieved from WF1 and use id and createdBy attributes if present
-        observed_daily = next(
-            (daily for daily in grouped_dailies[forecast.station_code] if daily.utcTimestamp == forecast_timestamp), None)
+        observed_daily = next((daily for daily in grouped_dailies[forecast.station_code] if daily.utcTimestamp == forecast_timestamp), None)
         forecast_id = observed_daily.forecast_id if observed_daily is not None else None
-        created_by = observed_daily.created_by if observed_daily is not None else None
+        created_by = username
```

Comment:
We're potentially patching an existing forecast. I wonder if we should retain the original `created_by` value and set the `updated_by` field?

Reply:
Thanks! Updated in https://github.com/bcgov/wps/pull/3824/commits/c17da1cf4820f6b6c9e4168220afb6e81a2b9313

Reply:
Do you think we should try setting `lastModifiedBy` to `username`?

Reply:
Unfortunately it and `createdBy` get overwritten with our token details.